emcc.py

```diff
@@ -1195,9 +1186,22 @@ def get_last_setting_change(setting):
       # we also do not support standalone mode in fastcomp.
       shared.Settings.STANDALONE_WASM = 1
 
-    if options.no_entry or ('_main' not in shared.Settings.EXPORTED_FUNCTIONS and
-                            '__start' not in shared.Settings.EXPORTED_FUNCTIONS):
+    if options.no_entry:
       shared.Settings.EXPECT_MAIN = 0
+    elif shared.Settings.STANDALONE_WASM:
+      if '_main' in shared.Settings.EXPORTED_FUNCTIONS:
+        diagnostics.warning('export-main', 'including `_main` in EXPORTED_FUNCTIONS is not necessary in standalone mode')
+    else:
+      # In normal non-standalone mode we have special handling of `_main` in EXPORTED_FUNCTIONS.
```

Comment:
what about non-standalone but with `--no-entry`?

Reply:
The `--no-entry` case is handled above in the `if` case.   Its the same for both standalone and normal mode, its just sets `shared.Settings.EXPECT_MAIN = 0` and leaves EXPORTED_FUNCTIONS as empty by default.